datafusion/common/src/error.rs

```diff
@@ -411,6 +399,56 @@ impl DataFusionError {
     }
 }
 
+/// Unwrap an `Option` if possible. Otherwise return an `DataFusionError::Internal`.
+/// In normal usage of DataFusion the unwrap should always succeed.
+///
+/// Example: `let values = unwrap_or_internal_err!(values)`
+#[macro_export]
+macro_rules! unwrap_or_internal_err {
+    ($Value: ident) => {
+        $Value.ok_or_else(|| {
+            DataFusionError::Internal(format!(
+                "{} should not be None",
+                stringify!($Value)
+            ))
+        })?
+    };
+}
+
+#[macro_export]
+macro_rules! with_dollar_sign {
+    ($($body:tt)*) => {
+        macro_rules! __with_dollar_sign { $($body)* }
+        __with_dollar_sign!($);
+    }
+}
```

Comment:
Hmm, does it define a macro `__with_dollar_sign` every time `with_dollar_sign` is called? If we call `make_error` multiple times, does it conflict?

Reply:
it doesn't conflict, but if you expand macros in 
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e5aad70ec87a420a6d4c9a7ba023c56d

you can see `__with_dollar_sign` created twice, but since the macro is not exported it is okay

Reply:
Interesting, I thought the macros with same name even not exposed will conflict each other.